tests/Cosmonaut.System/CosmosQueryInterceptionTests.cs

```diff
@@ -0,0 +1,187 @@
+﻿using Cosmonaut.System.Models;
+using FluentAssertions;
+using Microsoft.Azure.Documents.Client;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Cosmonaut.System
+{
+    public class CosmosQueryInterceptionTests : IDisposable
+    {
+        private readonly ICosmonautClient _cosmonautClient;
+        private readonly Uri _emulatorUri = new Uri("https://localhost:8081");
+        private readonly string _databaseId = $"DB{nameof(CosmosStoreTests)}";
+        private readonly string _collectionName = $"COL{nameof(CosmosStoreTests)}";
+        private readonly string _emulatorKey =
+            "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==";
+
+        private readonly ConnectionPolicy _connectionPolicy = new ConnectionPolicy
+        {
+            ConnectionProtocol = Protocol.Tcp,
+            ConnectionMode = ConnectionMode.Direct
+        };
+
+        public CosmosQueryInterceptionTests()
+        {
+            _cosmonautClient = new CosmonautClient(_emulatorUri, _emulatorKey, _connectionPolicy);
+            Seed().GetAwaiter().GetResult();
+        }
+
+        [Fact]
+        public void WhenInteceptorIsAppliedToQuery_ThenResultsShouldHaveOneItem()
```

Comment:
Please use `async Task` and use the awaitable extensions below, instead of the synchronous `.ToList()` ones.